compaction/table_state.hh

```diff
@@ -51,6 +53,7 @@ public:
     virtual const tombstone_gc_state& get_tombstone_gc_state() const noexcept = 0;
     virtual compaction_backlog_tracker& get_backlog_tracker() = 0;
     virtual const std::string& get_group_id() const noexcept = 0;
+    virtual seastar::condition_variable& get_staging_condition() noexcept = 0;
```

Comment:
This name doesn't tell anything about when this condition is fulfilled.

Reply:
The condition is checked by the caller's predicate.
The condition variable is just the means to signal blocked fibers that something might have changed and they should check the condition again.

Reply:
That said, I can make the name more meaningful, so it will be more obvious what it is used for.